engine/src/main/java/io/zeebe/engine/nwe/container/SubProcessProcessor.java

```diff
@@ -57,13 +57,27 @@ public void onActivating(
   public void onActivated(
       final ExecutableFlowElementContainer element, final BpmnElementContext context) {
 
-    final var noneStartEvent = element.getNoneStartEvent();
-    if (noneStartEvent == null) {
-      throw new BpmnProcessingException(
-          context, "Expected to activate the none start event of the sub-process but not found.");
-    }
+    if (element.hasNoneStartEvent()) {
+      // embedded sub-process is activated
+      final var noneStartEvent = element.getNoneStartEvent();
+      if (noneStartEvent == null) {
```

Comment:
can this still be null?

Reply:
It should not. I can remove the `null` checks.